src/shared/defaultAwsContext.ts

```diff
@@ -7,6 +7,10 @@ import * as AWS from 'aws-sdk'
 import * as vscode from 'vscode'
 import { AwsContext, AwsContextCredentials, ContextChangeEventsArgs } from './awsContext'
 import { regionSettingKey } from './constants'
+import { getLogger } from '../shared/logger'
+
+const logged = new Set<string>()
```

Comment:
TODO (future, out of scope in this PR): we could add a `logOnce()` to the Logger interface since this is a common pattern. Or I wonder if we can use something like the `debounce` library to do this...